packages/ui/src/composables/useResizeObserver.ts

```diff
@@ -0,0 +1,33 @@
+import { onBeforeUnmount, onMounted, Ref, unref, watch } from 'vue'
+
+type MaybeRef<T> = T | Ref<T>
+
+/**
+ * Resize observer hook
+ * @param elementsList {(HTMLElement | Ref<HTMLElement>)[]} List of HTML elements
```

Comment:
Wrong types here. Anyway, you don't need to type them manually, because you have typescript types.